app/Models/Workshop.php

```diff
@@ -69,6 +70,23 @@ public function externs()
         });
     }
 
+    /**
+     * Returns all the balances the workshop has had in different semesters.
+     */
+    public function balances(): HasMany
+    {
+        return $this->hasMany(WorkshopBalance::class);
+    }
+
+    /**
+     * Returns the balance for a given semester
+     * (by default the current one).
+     */
+    public function balance(int $semester = null): WorkshopBalance
+    {
+        return $this->balances()->where('semester_id', $semester ?? Semester::current()->id)->firstOrFail();
```

Comment:
Won't this die if we switch to next semester and the current workshop balances are not generated yet?
I'd allow null value and handle that.